elide-dbmanager/elide-dbmanager-hibernate5/src/main/java/com/yahoo/elide/dbmanagers/hibernate5/PersistenceManager.java

```diff
@@ -0,0 +1,102 @@
+package com.yahoo.elide.dbmanagers.hibernate5;
+
+import com.yahoo.elide.core.DatabaseManager;
+import com.yahoo.elide.core.DatabaseTransaction;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.security.User;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.metamodel.EntityType;
+import java.io.IOException;
+
+/**
+ * Manager for javax.persistence compatible db resource
+ */
+public class PersistenceManager extends DatabaseManager {
+    private final EntityManagerFactory entityManagerFactory;
+
+    public PersistenceManager(EntityManagerFactory entityManagerFactory) {
+        this.entityManagerFactory = entityManagerFactory;
+    }
+
+    @Override
+    public void populateEntityDictionary(EntityDictionary dictionary) {
+        for (EntityType entity : entityManagerFactory.getMetamodel().getEntities()) {
```

Comment:
What if `getMetaModel()` or `getEntities()` returns `null`? Should we throw something more descriptive here than `NPE`?